RetailCoder.VBE/Refactorings/Rename/RenameRefactoring.cs

```diff
@@ -38,358 +40,496 @@ public void Refactor()
             var presenter = _factory.Create();
             _model = presenter.Show();
 
-            QualifiedSelection? oldSelection = null;
+            QualifiedSelection? qOldSelection = null;
             var pane = _vbe.ActiveCodePane;
-            var module = pane.CodeModule;
             if (!pane.IsWrappingNullReference)
             {
-                oldSelection = module.GetQualifiedSelection();
+                qOldSelection = pane.CodeModule.GetQualifiedSelection();
             }
 
-            if (_model?.Declarations != null)
-            {
-                Rename();
-            }
+            Rename();
 
-            if (oldSelection.HasValue)
+            if (qOldSelection.HasValue)
             {
-                pane.Selection = oldSelection.Value.Selection;
+                pane.Selection = qOldSelection.Value.Selection;
             }
         }
 
-        public void Refactor(QualifiedSelection target)
+        //The User has selected something in a code pane
+        public void Refactor(QualifiedSelection qSelection)
         {
             var pane = _vbe.ActiveCodePane;
             if (pane.IsWrappingNullReference)
             {
                 return;
             }
-            pane.Selection = target.Selection;
+            pane.Selection = qSelection.Selection;
             Refactor();
         }
 
+        //The refactor starts from a Declaration only...not sure where the user selection is(?)
         public void Refactor(Declaration target)
         {
-            if (!target.IsUserDefined) { return; }
+            if (!target.IsUserDefined)
+            {
+                //Todo: Add format string to UI resources
+                var format = "Unable to rename Built-In declarations like '{0}'";
```

Comment:
Given how we've reversed the `IsBuiltIn` flag to become `IsUserDefined` because what we really mean to separate is "user-defined things" vs "everything else", the "built-in" terminology isn't quite accurate - it's anything declared in any referenced type library, *including* the built-in references. But using "built-in" to describe them is misleading.

I think I'd just go with a straightforward:

> Identifier '{0}' cannot be renamed.

With the placeholder possibly giving the actual fully-qualified name of the declaration, so if you try to rename `MsgBox` you get:

> Identifier 'VBA.Interaction.MsgBox' cannot be renamed.

Thoughts?